lib/renderers/cairo.h

```diff
@@ -155,11 +155,8 @@ bm_cairo_draw_line(struct cairo *cairo, struct cairo_paint *paint, struct cairo_
     pango_cairo_show_layout(cairo->cr, layout);
 
     if (paint->draw_cursor) {
-        size_t chr = 0;
-        for (size_t c = 0; c < paint->cursor; ++chr, c += bm_utf8_rune_next(buffer, c));
-
         PangoRectangle rect;
-        pango_layout_index_to_pos(layout, chr, &rect);
+        pango_layout_index_to_pos(layout, paint->cursor, &rect);
 
```

Comment:
This most likely will break multibyte text. (CJK)

Reply:
Note that chr is byte index.

Reply:
Or isn't, I can't read. Weird, are you sure this takes byte index? The documentation seems to talk about "grapheme".
>Converts from an index within a PangoLayout to the onscreen position corresponding to the grapheme at that index

Reply:
>index_ | byte index within layout

Seems so.. I remember this not working with simply `paint->cursor` though :S
Can you test with some CJK text?

Reply:
CJK text seems to work. I don't have a good way of typing these characters so I tested with `bm_menu_set_filter` and `xdotool type`.